modules/speech-to-text-workflowoperation/src/main/java/org/opencastproject/workflow/handler/speechtotext/SpeechToTextWorkflowOperationHandler.java

```diff
@@ -125,20 +129,53 @@ public WorkflowOperationResult start(WorkflowInstance workflowInstance, JobConte
           throws WorkflowOperationException {
 
     MediaPackage mediaPackage = workflowInstance.getMediaPackage();
+    WorkflowOperationInstance operation = workflowInstance.getCurrentOperation();
     logger.info("Start speech-to-text workflow operation for media package {}", mediaPackage);
 
+
     ConfiguredTagsAndFlavors tagsAndFlavors = getTagsAndFlavors(workflowInstance,
             Configuration.none, Configuration.one,
             Configuration.many, Configuration.one);
-    MediaPackageElementFlavor sourceFlavor = tagsAndFlavors.getSingleSrcFlavor();
+    List<MediaPackageElementFlavor> sourceFlavor = tagsAndFlavors.getSrcFlavors();
```

Comment:
Can use `.getSingleTargetFlavor()` here

Reply:
Agree. Seems like the next lines check for and only take the first flavor anyway.
I think `Configuration.one` should also make the check for one flavor unnecessary, shouldn't it?

Reply:
I don't understand what you mean with using `configuration.many`

The method is (Workflow instance, Src Tags, Src Flavors, Target Tags, Target Flavors). 

`Configuration.one` takes the first Flavor only.

How do you plan to use `.getSingleSourceFlavor()`? is not a valid method for that type of object.

Reply:
My bad, it's called `getSingleSrcFlavor`. See other comment below